### ISSUE TITLE
incorrect "redundant bracket" hint

### ISSUE THREAD

**Issue by MangoIV** [ndmitchell/hlint#1501]
HLint hints that I should remove the parens here: 
```haskell
.. \(Proxy @a) -> ...
```
this would be a parsing error.

**Comment by zliu41** [comment]
Fixed in https://github.com/ndmitchell/hlint/pull/1519
